src/store/index.ts

```diff
@@ -24,3 +25,23 @@ export const useExperienceSettingsState = () => {
 
   return [experienceSettings, handleSetExperienceSettings];
 };
+
+const guildState = atom({
+  key: "guildState",
+  default: JSON.parse(localStorage.getItem("guildState") || "[]"),
+});
+
+export const useGuildState = () => {
+  const [guilds, setGuilds] = useRecoilState(guildState);
+
+   // eslint-disable-next-line @typescript-eslint/no-explicit-any
+   const handleSetGuilds = (newGuilds: TGuild[]) => {
+    localStorage.setItem(
```

Comment:
Interesting that you've used localStorage here however making a call for the very first time is ok.

Anyway, it is great that you've implemented this.